livelike/_attribution.py

```diff
@@ -0,0 +1,351 @@
+import re
+
+import numpy as np
+import pandas as pd
+
+
+def columns_to_labels(
+    df: pd.DataFrame,
+    filter: str,
+    scrub: None | str = None,
+    keep_index: bool = True,
+) -> pd.DataFrame:
+    """
+    Converts dummy columns to labels.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Data frame containing dummy columns.
+    filter : str
+        A regular expression for filtering the columns to
+        use for labeling.
+    scrub : str
+        A regular expression for label text to exclude.
+    keep_index : bool
+        Whether to keep the input data frame index for
+        the output labels
+
+    Returns
+    -------
+    labels : pandas.Series
+        Converted labels.
+    """
+    df = df.loc[:, df.columns.str.contains(filter)]
+
+    df_ = pd.DataFrame()
+    for col in df.columns:
+        df_[col] = np.where(df[col] > 0, col, "")
+
+    labels = df_.sum(axis=1)
+    if scrub is not None:
+        labels = labels.apply(lambda x: re.sub(scrub, "", x))
+
+    if keep_index:
+        labels.index = df.index
+
+    return labels
+
+
+## Labeling Functions ##
+
+
+def class_of_worker(est_person: pd.DataFrame):
+    """
+    Produces person-level labels for class of worker.
+
+    Parameters
+    ----------
+    est_person: pandas.DataFrame
+        Person-level PUMS constraints.
+
+    Returns
+    -------
+    labels : pandas.Series
+        Class of worker labels.
+    """
+    filter = "^sexcw_male_|^sexcw_female_"
```

Comment:
Same as above (but for variables).

[Pre-commit is failing](https://results.pre-commit.ci/run/github/951924054/1746721171.Kee-9ujNQZCo2GDAWhIwCA) becauue `filter` variable shadows the Python [builtin `filter` function](https://docs.python.org/3/library/functions.html#filter)

```
livelike/_attribution.py:67:5: A001 Variable `filter` is shadowing a Python builtin
   |
65 |         Class of worker labels.
66 |     """
67 |     filter = "^sexcw_male_|^sexcw_female_"
   |     ^^^^^^ A001
68 |     labels = columns_to_labels(est_person, filter=filter, scrub=filter)
69 |     labels.name = "class_of_worker"
   |
```

Same rename as `filter_regex` or similar would do the trick to make the code more robust.

This applies to all declarations of `filter` here.